### PR TITLE
fix: UI freezes on enabling histogram

### PR DESCRIPTION
**Pull request by omkarK06**
#4806 

Issue replication steps:
Go to logs, turn off histogram, select stream and click run-query after getting results, turn on histogram.

## Summary by CodeRabbit

- **New Features**
	- Introduced functionality for refreshing data based on user settings.
	- Added dynamic SQL query management capabilities.
	- Added a method for generating histogram data skeletons.
	- Updated the region data fetching process for improved structure.

- **Improvements**
	- Enhanced error handling with detailed messages and better trace ID management.
	- Improved pagination management and histogram data handling.
	- Updated component structure for better search query handling and visualizations.
	- Enhanced clarity and maintainability of data filtering logic in the permission table.

- **Minor Adjustments**
	- Various code adjustments for clarity and efficiency.

### CODE REVIEW

**Review by hengfeiyang — APPROVED** [review]

File: web/src/plugins/logs/Index.vue [hunk 1]
```diff
@@ -794,12 +795,12 @@ export default defineComponent({
...
               searchObj.data.query = searchObj.data.query.replace(
                 "[FIELD_LIST]",
-                searchObj.data.stream.interestingFieldList.join(",")
+                searchObj.data.stream.interestingFieldList.join(","),
               );
             } else {
               searchObj.data.query = searchObj.data.query.replace(
                 "[FIELD_LIST]",
-                "*"
+                "*",
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/plugins/logs/Index.vue:805]: Verified

File: web/src/plugins/logs/Index.vue [hunk 2]
```diff
@@ -899,7 +900,7 @@
             let filteredData = removeFieldByName(parsedSQL.columns, field.name);
 
             const index = searchObj.data.stream.interestingFieldList.indexOf(
-              field.name
+              field.name,
             );
             if (index > -1) {
               searchObj.data.stream.interestingFieldList.splice(index, 1);
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/plugins/logs/Index.vue:908]: Verified

File: web/src/plugins/logs/Index.vue [hunk 3]
```diff
@@ -1061,7 +1062,7 @@
       // if x axis fields length is 2, then add 2nd x axis field to breakdown fields
       if (dashboardPanelData.data.queries[0].fields.x.length == 2) {
         dashboardPanelData.data.queries[0].fields.breakdown.push(
-          dashboardPanelData.data.queries[0].fields.x[1]
+          dashboardPanelData.data.queries[0].fields.x[1],
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/plugins/logs/Index.vue:1067]: Verified

File: web/src/plugins/logs/Index.vue [hunk 4]
```diff
@@ -987,7 +988,7 @@
 
       if (errors.length) {
         showErrorNotification(
-          "There are some errors, please fix them and try again"
+          "There are some errors, please fix them and try again",
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/plugins/logs/Index.vue:993]: Verified

File: web/src/plugins/logs/Index.vue [hunk 5]
```diff
@@ -936,7 +937,7 @@
           .replace(/`/g, "")
           .replace(
             searchObj.data.stream.selectedStream[0],
-            `"${searchObj.data.stream.selectedStream[0]}"`
+            `"${searchObj.data.stream.selectedStream[0]}"`,
           );
         searchObj.data.query = newQuery;
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/plugins/logs/Index.vue:944]: Verified

File: web/src/composables/useLogs.ts [hunk 6]
```diff
@@ -1803,12 +1792,27 @@ const useLogs = () => {
...
+      chartParams: {
+        title: getHistogramTitle(),
+        unparsed_x_data: [],
+        timezone: "",
+      },
+      errorCode: 0,
+      errorMsg: "Histogram is not available for limit queries.",
+      errorDetail: "",
+    };
+  }
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/composables/useLogs.ts]: Updated

File: web/src/composables/useLogs.ts [hunk 7]
```diff
@@ -4224,12 +4231,15 @@
...
     getHistogramQueryData,
+    generateHistogramSkeleton,
     fnParsedSQL,
     addOrderByToQuery,
     getRegionInfo,
     validateFilterForMultiStream,
     cancelQuery,
     reorderSelectedFields,
+    resetHistogramWithLimitError,
+    isNonAggregatedQuery,
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/composables/useLogs.ts]: Added todo for same

File: web/src/plugins/logs/Index.vue [hunk 8]
```diff
@@ -459,12 +459,15 @@
...
       getHistogramQueryData,
+      generateHistogramSkeleton,
       fnParsedSQL,
       addOrderByToQuery,
       getRegionInfo,
       getStreamList,
       getFunctions,
       extractFields,
+      resetHistogramWithLimitError,
+      isNonAggregatedQuery,
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/plugins/logs/Index.vue]: getRegionInfo and getFunctions are used. Removed addOrderByToQuery from import

File: web/src/plugins/logs/Index.vue [hunk 9]
```diff
@@ -1099,25 +1102,25 @@
...
     watch(
       () => dashboardPanelData.data.type,
       async () => {
         // await nextTick();
         visualizeChartData.value = JSON.parse(
-          JSON.stringify(dashboardPanelData.data)
+          JSON.stringify(dashboardPanelData.data),
         );
-      }
+      },
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/plugins/logs/Index.vue:1114]: Fine for now

File: web/src/composables/useLogs.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/composables/useLogs.ts:4249]: Noted

**Pull request merged**